### ISSUE TITLE
JDBC Inconsistent Support for Postgres hstore array Type

### ISSUE THREAD

**Issue by johnallen3d** [jeremyevans/sequel#1053]
We are using JRuby with `jdbc-postgres` (9.4.1200) and are receiving an error (`Java::JavaSql::SQLFeatureNotSupportedException: Method org.postgresql.jdbc4.Jdbc4Array.getArrayImpl(long,int,Map) is not yet implemented.`) when trying to create records with a field of type hstore array (`hstore[]`) using `Model#create` or `Model#save`. However switching to `Model#insert` allows the insert to succeed. In addition when retrieving a record with an `hstore[]` we receive a similar error.

self contained example:

```ruby
require 'jdbc/postgres'
Jdbc::Postgres.load_driver

require 'sequel'

pg = Sequel.connect('jdbc:postgresql://localhost/postgres?user=postgres')

sql=<<-SQL
  DROP EXTENSION hstore;
  DROP DATABASE test;
  CREATE EXTENSION hstore;
  CREATE DATABASE test;
SQL

pg.run sql

DB = Sequel.connect('jdbc:postgresql://localhost/test?user=postgres')

DB.extension :pg_hstore
DB.extension :pg_array
Sequel.extension :pg_array_ops
DB.register_array_type('hstore')

DB.run 'CREATE TABLE foos (bar hstore[]);'

class Foo < Sequel::Model(DB); end

attributes = { bar: Sequel.pg_array([Sequel.hstore({})]) }

begin
  Foo.create(attributes)
rescue StandardError => error
  puts 'error using create'
  puts error.message
else
  puts 'success using create'
end

begin
  Foo.new(attributes).save
rescue StandardError => error
  puts 'error using save'
  puts error.message
else
  puts 'success using save'
end

begin
  Foo.insert(attributes)
rescue StandardError => error
  puts 'error using insert'
  puts error.message
else
  puts 'success using insert'
end

begin
  Foo.first
rescue StandardError => error
  puts 'error selecting'
  puts error.message
else
  puts 'success selecting'
end

```

output:

```sh
> ruby spec/sequel_pg_array_error.rb

error using create
Java::JavaSql::SQLFeatureNotSupportedException: Method org.postgresql.jdbc4.Jdbc4Array.getArrayImpl(long,int,Map) is not yet implemented.

error using save
Java::JavaSql::SQLFeatureNotSupportedException: Method org.postgresql.jdbc4.Jdbc4Array.getArrayImpl(long,int,Map) is not yet implemented.

success using insert

error selecting
Java::JavaSql::SQLFeatureNotSupportedException: Method org.postgresql.jdbc4.Jdbc4Array.getArrayImpl(long,int,Map) is not yet implemented.
```

full backtrace:

```sh
Sequel::DatabaseError: Java::JavaSql::SQLFeatureNotSupportedException: Method org.postgresql.jdbc4.Jdbc4Array.getArrayImpl(long,int,Map) is not yet implemented.
                   raise_error at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/database/misc.rb:403
                     statement at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/adapters/jdbc.rb:604
              block in execute at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/adapters/jdbc.rb:259
                          hold at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/connection_pool/threaded.rb:91
                   synchronize at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/database/connecting.rb:240
                       execute at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/adapters/jdbc.rb:258
                       execute at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/dataset/actions.rb:860
                    fetch_rows at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/adapters/jdbc.rb:671
          returning_fetch_rows at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/dataset/actions.rb:939
                        insert at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/dataset/actions.rb:357
                        insert at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/adapters/shared/postgres.rb:1158
                 insert_select at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/adapters/shared/postgres.rb:1173
            _insert_select_raw at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/model/base.rb:1689
                       _insert at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/model/base.rb:1662
                block in _save at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/model/base.rb:1721
                 around_create at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/model/base.rb:972
                block in _save at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/model/base.rb:1718
                   around_save at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/model/base.rb:972
                         _save at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/model/base.rb:1713
                 block in save at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/model/base.rb:1390
  block in checked_transaction at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/model/base.rb:1862
                  _transaction at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/database/transactions.rb:128
          block in transaction at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/database/transactions.rb:103
                          hold at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/connection_pool/threaded.rb:104
                   synchronize at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/database/connecting.rb:240
                   transaction at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/database/transactions.rb:96
           checked_transaction at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/model/base.rb:1862
                 block in save at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/model/base.rb:1390
          checked_save_failure at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/model/base.rb:1850
                          save at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/model/base.rb:1390
                        create at /usr/local/bundle/gems/sequel-3.48.0/lib/sequel/model/base.rb:168
                         <top> at spec/sequel_pg_array_error.rb:30
```

**Comment by jeremyevans** [comment]
Thank you for the detailed explanation and self contained example.  Unfortunately, I can't reproduce in my environment.  I added some extra logging of the result of the create/new/insert/first calls:

```
$ jruby -I lib -r sequel ../jdbc-postgres-test.rb
#<Foo @values={:bar=>[{}]}>
success using create
#<Foo @values={:bar=>[{}]}>
success using save
nil
success using insert
#<Foo @values={:bar=>[{}]}>
success selecting
```

This is using the current master branch.  Since you are using sequel-3.48.0, I assume this is a bug that has since been fixed (maybe in 4.1.0, but I'm not sure).  So this should be fixed if you upgrade to the latest version of Sequel.

**Comment by johnallen3d** [comment]
I appreciate the quick response! Upgrading to the latest version resolved my issues. I guess I could have saved everyone some time by trying that first. :flushed:
